benchmark/README.md

```diff
@@ -0,0 +1,38 @@
+Microbenchmarks
+===============
+This directory contains the definition of a program that measures the timing and
+resource consumption of a test tracing scenario.
+
+The benchmark uses [Google Benchmark][1], whose source is included as a git
+submodule under [./google-benchmark][2].
+
+The scenario that's measured is similar to the [../example][3] setup.  A trace
+is created whose structure reflects that of a particular file directory
+structure.  The directory structure, in this case, is the source tree of the
+[Tiny C Compiler][4], whose source is included as a git submodule under
+[./tinycc][5].
+
+The scenario does not use the network, spawn any threads, or read/write
+any files. The operations that are implicitly covered by the scenario are:
+
+- configuring and initializing a tracer,
+- creating a trace,
+- adding spans to a trace,
+- setting tags on spans,
+- finishing spans,
+- finalizing a trace and making a sampling decision,
+- serializing a trace as MessagePack.
+
```

Comment:
Do we want a benchmark in the future that also does extraction or injection?
Not that they're heavyweight tasks at all, just for a form of completeness

Reply:
That's a good idea. Injection certainly. With extraction there's a trade-off between "we create a trace ID and calculate a sampling decision" and "we parsed the trace ID and sampling priority from request headers."

I suppose we could do both.

Will leave as a future extension.